esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -522,6 +525,38 @@ void EsdlServiceImpl::handleServiceRequest(IEspContext &context,
     const char *mthName = mthdef.queryName();
     context.addTraceSummaryValue("method", mthName);
 
+    StringBuffer trxid;
+    if (!m_bGenerateLocalTrxId)
+    {
+        if (m_oLoggingManager)
+        {
+            StringBuffer wsaddress;
+            short int port;
+            context.getServAddress(wsaddress, port);
+            VStringBuffer uniqueId("%s:%d-%u", wsaddress.str(), port, (unsigned) (memsize_t) GetCurrentThreadId());
+
+            StringAttrMapping trxidbasics;
+            StringBuffer creationTime;
+            creationTime.setf("%u", context.queryCreationTime());
+            trxidbasics.setValue("TransactionDateTime", creationTime.str());
+            trxidbasics.setValue("TransactionMethod", mthName);
+            trxidbasics.setValue("TransactionESPIP", uniqueId.str()); // this should be unique ID
```

Comment:
Is it possible to use the #define's in loggingagentbase.hpp to replace those hard coded strings?